toplevel/coqloop.ml

```diff
@@ -405,7 +405,17 @@ let rec vernac_loop ~state =
 
     | Some (VernacShowGoal {gid; sid}) ->
       let proof = Stm.get_proof ~doc:state.doc (Stateid.of_int sid) in
-      Feedback.msg_notice (Printer.pr_goal_emacs ~proof gid sid);
+      let goal = Printer.pr_goal_emacs ~proof gid sid in
+      let evars =
+          match proof with
+          | None -> mt()
+          | Some p ->
+            let gl = (Evar.unsafe_of_int gid) in
+            let Proof.{goals; stack; shelf; given_up; sigma} = Proof.data p in
```

Comment:
Don't declare variables that you are not using, also modern OCaml prefers internal record qualification I think:
```
let { Proof.sigma } = Proof.data p in ...
```

Reply:
Like many things, it's not easy to find OCaml language details in the OCaml doc.

Reply:
Indeed, coding style choices are mostly transmitted by folk knowledge.

Reply:
By the way the construct is record pattern matching which is fairly standard, documented here: https://caml.inria.fr/pub/docs/manual-ocaml/patterns.html but yeah that's not very user friendly.

Real world OCaml does a bit better http://dev.realworldocaml.org/records.html#patterns-and-exhaustiveness

Reply:
Thanks.

Though if there's more than one field you would write:

```
  let Proof.{ sigma; goals } = Proof.data p in
```

or 

```
  let { Proof.sigma; Proof.goals } = Proof.data p in

```

as you did in Pfedit.get_nth_V82_goal?

Reply:
```
let { Proof.sigma; goals } = ... 
```
is a commonly accepted way, once OCaml resolves a record projection it can do resolution for the rest.